test/e2e/federated-secret.go

```diff
@@ -98,19 +97,17 @@ func updateSecretOrFail(clientset *federation_release_1_4.Clientset, namespace s
 		Fail(fmt.Sprintf("Internal error: invalid parameters passed to updateSecretOrFail: clientset: %v, namespace: %v", clientset, namespace))
 	}
 
-	var err error
 	var newSecret *v1.Secret
-	secret := &v1.Secret{
-		ObjectMeta: v1.ObjectMeta{
-			Name: UpdatedFederatedSecretName,
-		},
-	}
-
 	for retryCount := 0; retryCount < MaxRetries; retryCount++ {
-		_, err = clientset.Core().Secrets(namespace).Get(FederatedSecretName)
+		secret, err := clientset.Core().Secrets(namespace).Get(FederatedSecretName)
 		if err != nil {
 			framework.Failf("failed to get secret %q: %v", FederatedSecretName, err)
 		}
+
+		// Update the secret by adding a label.
+		secret.ObjectMeta.Labels = map[string]string{
```

Comment:
nit: It might be worth changing this to an update of the Data field, which is probably the most common kind of update.